source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/wrappers/torchrl/torchrl_env_wrapper.py

```diff
@@ -0,0 +1,601 @@
+            keys = info_dict.keys()
+            if self.ignore_private:
+                keys = [key for key in keys if not key.startswith("_")]
+            self.keys = keys
+        # create an info_spec only if there is none
+        info_spec = None if self.info_spec is not None else CompositeSpec()
+
+        def add_value_to_tensordict(tensordict: TensorDict, key: str, value: torch.Tensor):
+            if value.dtype == np.dtype("O"):
+                value = np.stack(value)
+            # if value.shape != tensordict.shape:
+            value = value.expand(tensordict.shape)
+            tensordict.set(key, value)
+            if info_spec is not None:
+                value = tensordict.get(key)
+                info_spec[key] = UnboundedContinuousTensorSpec(value.shape, device=value.device, dtype=value.dtype)
+
+        for key in keys:
+            if key in info_dict:
+                val = info_dict[key]
+                if isinstance(val, dict):
+                    # Add all key strings that map to dictionaries to the tensordict.
+                    # This is done because sampling throughput is reduced
+                    # if the tensordict key set does not match the expected.
+                    tensordict.set(key, torch.tensor(0).expand(tensordict.shape))
+                    nested_tensordict = TensorDict(batch_size=tensordict.shape, device=tensordict.device)
+                    key = key + "_data"
+                    for k, v in val.items():
+                        v = torch.as_tensor(v, device=tensordict.device)
+                        add_value_to_tensordict(nested_tensordict, k, v)
+                    tensordict.set(key, nested_tensordict)
+            elif info_spec is not None:
+                if key in info_spec:
+                    # Fill missing with 0s
+                    tensordict.set(key, info_spec[key].zero())
+            else:
+                raise KeyError(f"The key {key} could not be found or inferred.")
+        if info_spec is not None:
+            if tensordict.device is not None:
+                info_spec = info_spec.to(tensordict.device)
+            self._info_spec = info_spec
+        # add episode length field to tensordict for logging
+        tensordict.set("episode_length", torch.tensor(0.0).expand(tensordict.shape))
+        tensordict.set("episode_reward", torch.tensor(0.0).expand(tensordict.shape))
+        return tensordict
+
+
+class SyncDataCollectorWrapper(SyncDataCollector):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def iterator(self) -> Iterator[TensorDictBase]:
+        """Iterates through the DataCollector.
+
+        Yields: TensorDictBase objects containing (chunks of) trajectories
+
+        """
+        # The portion of the code handling cuda streams has been removed in this inherited method, which
+        # caused CUDA memory allocation issues with IsaacSim during env stepping.
+        total_frames = self.total_frames
```

Comment:
gotcha I think we can solve this by providing a `no_stream` arg in the collector - happy to make it a feature if you think that would help